### PR TITLE
RFE add search functionality to the mini report dialog

### LINKED ISSUE #4003: RFE add search functionality to the mini report dialog
RFE add search functionality to the mini report dialog

search through the currently selected report tab based on the chosen options

search by players
search by owned units
search by keyword

### PR DESCRIPTION
**Pull request by kuronekochomusuke**
RFE #4003 add search functionality to the mini report dialog

Edit: Fixes #4003 

search through the currently selected report tab based on the chosen options

search by players
search by owned units
search by keyword

image

image

### CODE REVIEW

**Comment by HammerGS** [comment]
I think this has a lot of possibilities and will play around with it this weekend.

File: megamek/src/megamek/client/ui/swing/util/UIUtil.java [hunk 1]
```diff
@@ -1187,4 +1187,63 @@ public static void updateWindowBounds(Window window) {
...
+        else {
+            currentPos = startPos - searchPattern.length() - 1;
+
+            if (currentPos < 1) {
+                currentPos = len;
+            }
+        }
+
+        while (startPos != currentPos) {
+            String match = text.substring(currentPos, currentPos + searchPattern.length());
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/util/UIUtil.java:1213]: Can't you just replace this whole thing with text.indexOf(searchPattern, startindex)? I'm all for rolling your own algorithms, but this is a solved problem, and I don't think we need a home-grown O(n*m) solution here.

Comment (NickAragua) [reply megamek/src/megamek/client/ui/swing/util/UIUtil.java:1213]: Searching backwards can be accomplished by using lastIndexOf(searchPattern, startIndex)

**Review by NickAragua — CHANGES_REQUESTED** [review]
Needs a little less home-grown O(n*m) text search algorithm, but good idea otherwise.

**Comment by kuronekochomusuke** [comment]
added the hyperlink support

**Review by NickAragua — APPROVED** [review]

**Pull request merged**